src/main/java/io/kestra/plugin/apify/task/AbstractGetDataset.java

```diff
@@ -0,0 +1,171 @@
+
+    @Schema(
+        title = "Flatten",
+        description = "List of fields which should transform nested objects into flat structures. For example, with " +
+            "flatten=\"foo\" the object {\"foo\":{\"bar\": \"hello\"}} is turned into {\"foo.bar\": \"hello\"}."
+    )
+    @Builder.Default
+    private Property<Boolean> flatten = Property.ofValue(false);
+
+    @Schema(
+        title = "sort",
+        description = "Sort the runs by startedAt in descending order. Defaults to `ASC`."
+    )
+    @Builder.Default
+    private Property<ApifySortDirection> sort = Property.ofValue(ApifySortDirection.ASC);
+
+    @Schema(
+        title = "SkipEmpty",
+        description = "If true then empty items are skipped from the output. Default value is true."
+    )
+    @Builder.Default
+    private Property<Boolean> skipEmpty = Property.ofValue(true);
+
+    @Schema(
+        title = "SkipFailedPages",
+        description = "If true then, the all the items with errorInfo property will be skipped from the output. Default value false."
+    )
+    @Builder.Default
+    private Property<Boolean> skipFailedPages = Property.ofValue(false);
+
+    @Schema(
+        title = "View",
+        description = "Defines the view configuration for dataset items based on the schema definition. This " +
+            "parameter determines how the data will be filtered and presented. For complete specification details, " +
+            "see the dataset schema documentation in the Apify documentation."
+    )
+    private Property<String> view;
+
+    @Schema(
+        title = "SkipHidden",
+        description = "If true then hidden fields are skipped from the output, i.e. fields starting with the # character."
+    )
+    @Builder.Default
+    private Property<Boolean> skipHidden = Property.ofValue(false);
+
+
+    @Schema(
+        title = "Simplified",
+        description = "If true then hidden fields are skipped from the output, i.e. fields starting with the # character."
+    )
+    @Builder.Default
+    private Property<Boolean> simplified = Property.ofValue(false);
+
+    @Builder.Default
+    protected Duration DEFAULT_TIMEOUT_DURATION = Duration.ofSeconds(300);
+
+    public String buildURL(RunContext runContext) throws IllegalVariableEvaluationException {
+        String rDatasetId = runContext.render(this.datasetId).as(String.class).orElseThrow(
+            () -> new IllegalArgumentException("datasetId is required")
+        );
+
+        List<String> rUnwind = runContext.render(this.unwind).asList(String.class);
+        List<String> rFields = runContext.render(this.fields).asList(String.class);
+        List<String> rOmit = runContext.render(this.omit).asList(String.class);
+        Optional<String> rView = runContext.render(this.view).as(String.class);
+
+        final Map<String, Object> queryParamValues = new HashMap<>(Map.of(
+            "cleanValue", runContext.render(this.clean).as(Boolean.class).orElseThrow(),
```

Comment:
You can add default values in `orElse()` like this if any.
```suggestion
            "cleanValue", runContext.render(this.clean).as(Boolean.class).orElse(false),
```

Reply:
I've pushed the suggested changes. I was thinking  if we use the `orElse()` here do we still need to use the default builder pattern? If all of the values that have default values already use the default builder pattern then it is not possible for the value in the 'orElse' to be assigned to the local variable. Since all of the input in these tasks are immutable objects could we add a final static variable to define the default value for specific class properties rather than using the default builder pattern?

For example:
``` java
class Join extends Task implements RunnableTask<Join.Output> {
    @NotNull
    Property<String> textOne;
    
    @NotNull
    Property<String> textTwo;
    
    Property<String> delimiter;
    
    private static final String DELIMITER = ",";
    
    @Override
    public Join.Output run(RunContext runContext) throws Exception {
        String rTextOne = runContext.render(textOne).as(String.class).orElseThrow(
            () -> new Exception("textOne is required")
        );
        String rTextTwo = runContext.render(textTwo).as(String.class).orElseThrow(
            () -> new Exception("textTwo is required")
        );
        String rDelimited = runContext.render(delimiter).as(String.class).orElse(DELIMITER);
        
        return new Join.Output(rTextOne + rDelimited + rTextTwo);
    }

    record Output(String text) implements io.kestra.core.models.tasks.Output {
    }
}
```

Reply:
You can mix the default builder annotation + the `orElse` as a more defensive approach + the const to put at both places yes.